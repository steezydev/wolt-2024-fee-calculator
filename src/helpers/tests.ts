const localStorageMock = (function () {
    let store: Record<string, string> = {};
    return {
        getItem(key: string) {
            return store[key] || null;
        },
        setItem(key: string, value: string) {
            store[key] = value.toString();
        },
        clear() {
            store = {};
        },
    };
})();

const matchMediaMock = jest.fn().mockImplementation((query) => ({
    matches: query === '(prefers-color-scheme: dark)',
    media: query,
    onchange: null,
    addListener: jest.fn(),
    removeListener: jest.fn(),
    addEventListener: jest.fn(),
    removeEventListener: jest.fn(),
    dispatchEvent: jest.fn(),
}));

export const mockContext = () => {
    beforeAll(() => {
        Object.defineProperty(window, 'localStorage', {
            value: {
                getItem: jest.fn().mockImplementation((key: string) => {
                    return localStorageMock.getItem(key);
                }),
                setItem: jest
                    .fn()
                    .mockImplementation((key: string, value: string) => {
                        localStorageMock.setItem(key, value);
                    }),
                clear: localStorageMock.clear,
            },
            writable: true,
        });

        window.matchMedia = matchMediaMock;
    });

    beforeEach(() => {
        (window.localStorage.getItem as jest.Mock).mockClear();
        (window.localStorage.setItem as jest.Mock).mockClear();
        matchMediaMock.mockClear();
    });
};

export const setupFakeDate = () => {
    beforeEach(() => {
        jest.useFakeTimers();
        jest.setSystemTime(new Date('2024-01-01'));
    });

    afterEach(() => {
        jest.useRealTimers();
    });
};

export const mockScrollTo = () => {
    beforeEach(() => {
        Element.prototype.scrollTo = jest.fn();
    });

    afterEach(() => {
        (Element.prototype.scrollTo as jest.Mock).mockRestore();
    });
};
