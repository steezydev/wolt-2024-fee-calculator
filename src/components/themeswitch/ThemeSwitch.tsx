import ButtonRound from '@/components/button/ButtonRound';
import MoonIcon from '@/components/icons/MoonIcon';
import SunIcon from '@/components/icons/SunIcon';
import { classNames } from '@/helpers/classNames';
import { ThemeSwitchProps } from '@/types/props/ThemeSwitchProps';
import { Theme } from '@/types/shared/Theme';

const ThemeSwitch = ({
  selectedTheme,
  onLightThemeClick,
  onDarkThemeClick,
  className,
}: ThemeSwitchProps) => {
  return (
    <div className={classNames('flex gap-3', className)}>
      <ButtonRound
        onClick={onLightThemeClick}
        id='light-theme-button'
        ariaLabel='Set light theme'
        className={classNames(
          selectedTheme === Theme.LIGHT ? 'bg-primary-300 text-white' : ''
        )}
      >
        <SunIcon />
      </ButtonRound>
      <ButtonRound
        onClick={onDarkThemeClick}
        id='dark-theme-button'
        ariaLabel='Set dark theme'
        className={classNames(
          selectedTheme === Theme.DARK ? 'bg-primary-300 text-white' : ''
        )}
      >
        <MoonIcon />
      </ButtonRound>
    </div>
  );
};

export default ThemeSwitch;
