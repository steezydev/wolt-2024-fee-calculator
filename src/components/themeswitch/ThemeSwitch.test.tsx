import ThemeSwitch from './ThemeSwitch';
import { themeSwitchRenderTests } from './ThemeSwitch.tests';

themeSwitchRenderTests('ThemeSwitch', ThemeSwitch);
