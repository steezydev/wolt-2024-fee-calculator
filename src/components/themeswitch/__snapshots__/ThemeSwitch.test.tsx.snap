// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ThemeSwitch when asked to render renders 1`] = `
<div
  className="flex gap-3"
>
  <button
    aria-label="Set light theme"
    className="rounded-full p-2 text-black dark:text-white hover:bg-primary-300/90 hover:text-white active:bg-primary-300/80 disabled:bg-primary-300/60 disabled:text-white/40 disabled:cursor-not-allowed focus-visible:ring-4 focus-visible:ring-focus focus:outline-none focus-visible:outline-none"
    data-test-id="light-theme-button"
    id="light-theme-button"
    onClick={[Function]}
  >
    <svg
      fill="none"
      height="24"
      stroke="currentColor"
      strokeLinecap="round"
      strokeLinejoin="round"
      strokeWidth="2"
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <circle
        cx="12"
        cy="12"
        r="4"
      />
      <path
        d="M12 2v2"
      />
      <path
        d="M12 20v2"
      />
      <path
        d="m4.93 4.93 1.41 1.41"
      />
      <path
        d="m17.66 17.66 1.41 1.41"
      />
      <path
        d="M2 12h2"
      />
      <path
        d="M20 12h2"
      />
      <path
        d="m6.34 17.66-1.41 1.41"
      />
      <path
        d="m19.07 4.93-1.41 1.41"
      />
    </svg>
  </button>
  <button
    aria-label="Set dark theme"
    className="rounded-full p-2 text-black dark:text-white hover:bg-primary-300/90 hover:text-white active:bg-primary-300/80 disabled:bg-primary-300/60 disabled:text-white/40 disabled:cursor-not-allowed focus-visible:ring-4 focus-visible:ring-focus focus:outline-none focus-visible:outline-none"
    data-test-id="dark-theme-button"
    id="dark-theme-button"
    onClick={[Function]}
  >
    <svg
      fill="none"
      height="24"
      stroke="currentColor"
      strokeLinecap="round"
      strokeLinejoin="round"
      strokeWidth="2"
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M12 3a6 6 0 0 0 9 9 9 9 0 1 1-9-9Z"
      />
    </svg>
  </button>
</div>
`;

exports[`ThemeSwitch when asked to render with additional class renders wtih additional class 1`] = `
<div
  className="flex gap-3 some-class"
>
  <button
    aria-label="Set light theme"
    className="rounded-full p-2 text-black dark:text-white hover:bg-primary-300/90 hover:text-white active:bg-primary-300/80 disabled:bg-primary-300/60 disabled:text-white/40 disabled:cursor-not-allowed focus-visible:ring-4 focus-visible:ring-focus focus:outline-none focus-visible:outline-none"
    data-test-id="light-theme-button"
    id="light-theme-button"
    onClick={[Function]}
  >
    <svg
      fill="none"
      height="24"
      stroke="currentColor"
      strokeLinecap="round"
      strokeLinejoin="round"
      strokeWidth="2"
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <circle
        cx="12"
        cy="12"
        r="4"
      />
      <path
        d="M12 2v2"
      />
      <path
        d="M12 20v2"
      />
      <path
        d="m4.93 4.93 1.41 1.41"
      />
      <path
        d="m17.66 17.66 1.41 1.41"
      />
      <path
        d="M2 12h2"
      />
      <path
        d="M20 12h2"
      />
      <path
        d="m6.34 17.66-1.41 1.41"
      />
      <path
        d="m19.07 4.93-1.41 1.41"
      />
    </svg>
  </button>
  <button
    aria-label="Set dark theme"
    className="rounded-full p-2 text-black dark:text-white hover:bg-primary-300/90 hover:text-white active:bg-primary-300/80 disabled:bg-primary-300/60 disabled:text-white/40 disabled:cursor-not-allowed focus-visible:ring-4 focus-visible:ring-focus focus:outline-none focus-visible:outline-none"
    data-test-id="dark-theme-button"
    id="dark-theme-button"
    onClick={[Function]}
  >
    <svg
      fill="none"
      height="24"
      stroke="currentColor"
      strokeLinecap="round"
      strokeLinejoin="round"
      strokeWidth="2"
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M12 3a6 6 0 0 0 9 9 9 9 0 1 1-9-9Z"
      />
    </svg>
  </button>
</div>
`;

exports[`ThemeSwitch when asked to render with dark theme selected renders with dark theme selected 1`] = `
<div
  className="flex gap-3"
>
  <button
    aria-label="Set light theme"
    className="rounded-full p-2 text-black dark:text-white hover:bg-primary-300/90 hover:text-white active:bg-primary-300/80 disabled:bg-primary-300/60 disabled:text-white/40 disabled:cursor-not-allowed focus-visible:ring-4 focus-visible:ring-focus focus:outline-none focus-visible:outline-none"
    data-test-id="light-theme-button"
    id="light-theme-button"
    onClick={[Function]}
  >
    <svg
      fill="none"
      height="24"
      stroke="currentColor"
      strokeLinecap="round"
      strokeLinejoin="round"
      strokeWidth="2"
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <circle
        cx="12"
        cy="12"
        r="4"
      />
      <path
        d="M12 2v2"
      />
      <path
        d="M12 20v2"
      />
      <path
        d="m4.93 4.93 1.41 1.41"
      />
      <path
        d="m17.66 17.66 1.41 1.41"
      />
      <path
        d="M2 12h2"
      />
      <path
        d="M20 12h2"
      />
      <path
        d="m6.34 17.66-1.41 1.41"
      />
      <path
        d="m19.07 4.93-1.41 1.41"
      />
    </svg>
  </button>
  <button
    aria-label="Set dark theme"
    className="rounded-full p-2 text-black dark:text-white hover:bg-primary-300/90 hover:text-white active:bg-primary-300/80 disabled:bg-primary-300/60 disabled:text-white/40 disabled:cursor-not-allowed focus-visible:ring-4 focus-visible:ring-focus focus:outline-none focus-visible:outline-none bg-primary-300 text-white"
    data-test-id="dark-theme-button"
    id="dark-theme-button"
    onClick={[Function]}
  >
    <svg
      fill="none"
      height="24"
      stroke="currentColor"
      strokeLinecap="round"
      strokeLinejoin="round"
      strokeWidth="2"
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M12 3a6 6 0 0 0 9 9 9 9 0 1 1-9-9Z"
      />
    </svg>
  </button>
</div>
`;

exports[`ThemeSwitch when asked to render with light theme selected renders with light theme selected 1`] = `
<div
  className="flex gap-3"
>
  <button
    aria-label="Set light theme"
    className="rounded-full p-2 text-black dark:text-white hover:bg-primary-300/90 hover:text-white active:bg-primary-300/80 disabled:bg-primary-300/60 disabled:text-white/40 disabled:cursor-not-allowed focus-visible:ring-4 focus-visible:ring-focus focus:outline-none focus-visible:outline-none bg-primary-300 text-white"
    data-test-id="light-theme-button"
    id="light-theme-button"
    onClick={[Function]}
  >
    <svg
      fill="none"
      height="24"
      stroke="currentColor"
      strokeLinecap="round"
      strokeLinejoin="round"
      strokeWidth="2"
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <circle
        cx="12"
        cy="12"
        r="4"
      />
      <path
        d="M12 2v2"
      />
      <path
        d="M12 20v2"
      />
      <path
        d="m4.93 4.93 1.41 1.41"
      />
      <path
        d="m17.66 17.66 1.41 1.41"
      />
      <path
        d="M2 12h2"
      />
      <path
        d="M20 12h2"
      />
      <path
        d="m6.34 17.66-1.41 1.41"
      />
      <path
        d="m19.07 4.93-1.41 1.41"
      />
    </svg>
  </button>
  <button
    aria-label="Set dark theme"
    className="rounded-full p-2 text-black dark:text-white hover:bg-primary-300/90 hover:text-white active:bg-primary-300/80 disabled:bg-primary-300/60 disabled:text-white/40 disabled:cursor-not-allowed focus-visible:ring-4 focus-visible:ring-focus focus:outline-none focus-visible:outline-none"
    data-test-id="dark-theme-button"
    id="dark-theme-button"
    onClick={[Function]}
  >
    <svg
      fill="none"
      height="24"
      stroke="currentColor"
      strokeLinecap="round"
      strokeLinejoin="round"
      strokeWidth="2"
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M12 3a6 6 0 0 0 9 9 9 9 0 1 1-9-9Z"
      />
    </svg>
  </button>
</div>
`;
